fv3core/fv3core/stencils/fillz.py

```diff
@@ -147,25 +152,34 @@ def make_storage(*args, **kwargs):
         self._sum0 = make_storage(shape_ij, origin=(0, 0))
         self._sum1 = make_storage(shape_ij, origin=(0, 0))
 
+        # [DaCe] linearlize tracers dict into a list at __init__ time
+        self._filtered_tracer_dict = {
+            name: tracers[name] for name in utils.tracer_variables[0 : self._nq]
+        }
+
+    @computepath_method
     def __call__(
         self,
         dp2: FloatField,
-        tracers: Dict[str, Any],
+        tracers: DaceConstant,
     ):
         """
         Args:
             dp2 (in): pressure thickness of atmospheric layer
             tracers (inout): tracers to fix negative masses in
         """
-        tracer_list = [tracers[name] for name in utils.tracer_variables[0 : self._nq]]
-        for tracer in tracer_list:
+        # [DaCe] runtime tracers is deactivate, was cached in __init__
+
+        # [DaCe] return value is not used
+        # [DaCe] dict.values/dict.items bug - using key indexing
+        for tracer_name in self._filtered_tracer_dict.keys():
             self._fix_tracer_stencil(
-                tracer,
+                tracers[tracer_name],
```

Comment:
Are we passing tracers in init because we need to know the list of tracer strings here? If that's the case, would it make sense to have the list of tracer names in config?